bin/sign.py

```diff
@@ -0,0 +1,308 @@
+"""A script which wraps ESRPClient.exe for code signing."""
+
+import json
+import os
+import subprocess
+import sys
+from argparse import ArgumentParser
+from argparse import ArgumentDefaultsHelpFormatter
+from argparse import Namespace
+from collections.abc import Iterator
+from collections.abc import Generator
+from contextlib import ExitStack
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any
+from zipfile import ZipFile
+from zipfile import ZIP_DEFLATED
+
+JSON = dict[str, Any]  # A naive type alias for JSON.
+
+
+def mac_app_developer_sign(key_code: str) -> JSON:
+    """Return the JSON for a `MacAppDeveloperSign` operation."""
+    return {
+        "KeyCode": key_code,
+        "OperationCode": "MacAppDeveloperSign",
+        "Parameters": {
+            "OpusName": "Microsoft",
+            "OpusInfo": "http://www.microsoft.com",
+            "FileDigest": '/fd "SHA256"',
+            "PageHash": "/NPH",
+            "TimeStamp": '/tr "http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer" /td sha256',
+        },
+        "ToolName": "sign",
+        "ToolVersion": "1.0",
+    }
+
+
+def sign_tool_sign(key_code: str) -> JSON:
+    """Return the JSON for a `SigntoolSign` operation."""
+    return {
+        "KeyCode": key_code,
+        "OperationCode": "SigntoolSign",
+        "Parameters": {
+            "OpusName": "Microsoft",
+            "OpusInfo": "http://www.microsoft.com",
+            "FileDigest": '/fd "SHA256"',
+            "PageHash": "/NPH",
+            "TimeStamp": '/tr "http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer" /td sha256',
+        },
+        "ToolName": "sign",
+        "ToolVersion": "1.0",
+    }

```

Comment:
```suggestion
def sign_core(key_code: str, op_code: str) -> JSON:
    """Return the JSON for a signing operation."""
    return {
        "KeyCode": key_code,
        "OperationCode": op_code,
        "Parameters": {
            "OpusName": "Microsoft",
            "OpusInfo": "http://www.microsoft.com",
            "FileDigest": '/fd "SHA256"',
            "PageHash": "/NPH",
            "TimeStamp": '/tr "http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer" /td sha256',
        },
        "ToolName": "sign",
        "ToolVersion": "1.0",
    }

def mac_app_developer_sign(key_code: str) -> JSON:
    """Return the JSON for a `MacAppDeveloperSign` operation."""
    return sign_core(key_code, "MacAppDeveloperSign")


def sign_tool_sign(key_code: str) -> JSON:
    """Return the JSON for a `SigntoolSign` operation."""
    return sign_core(key_code,"SigntoolSign")


```

Reply:
I didn't use this snippet verbatim because I preferred the name `sign_operation()` to `sign_core()` (I usually dislike `*core` names). Otherwise what I've updated is the same as your suggestion.